tests/Unit/StopwatchPluginTest.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Http\Client\Common\Plugin\Tests\Unit;
+
+use GuzzleHttp\Psr7\Request;
+use GuzzleHttp\Psr7\Response;
+use Http\Client\Common\Plugin\StopwatchPlugin;
+use Http\Client\Exception\HttpException;
+use Http\Promise\FulfilledPromise;
+use Http\Promise\RejectedPromise;
+use LogicException;
+use PHPUnit\Framework\TestCase;
+use Psr\Http\Message\RequestInterface;
+use Symfony\Component\Stopwatch\Stopwatch;
+
+final class StopwatchPluginTest extends TestCase
+{
+    /**
+     * @var Stopwatch
+     */
+    private $stopwatch;
+
+    /**
+     * @var StopwatchPlugin
+     */
+    private $plugin;
+
+    public function setUp(): void
+    {
+        $this->stopwatch = new Stopwatch();
+        $this->plugin = new StopwatchPlugin($this->stopwatch);
+    }
+
+    /**
+     * @test
+     */
+    public function it_records_event(): void
```

Comment:
the phpunit convention is to have methods start with `test` and then we would not need the annotation.

i don't know how much people use this form or the other form (but all things i work on have the `testXY` form). wdyt of renaming this to testRecordEvent, and the other testRecordError ?

Reply:
both are fine.. in our code base we only use it_should_be.... together with @test 
I decided to keep the original names from phpspec... given there are  just 2, does it really matter? :)

Reply:
no, does not really matter. just noticed and wanted to discuss it. fine for me.